kubectl-cloudflow/deploy/deploy.go

```diff
@@ -23,19 +27,9 @@ import (
 // GetCloudflowApplicationDescriptorFromDockerImage pulls a image and extracts the Cloudflow Application descriptor from a docker label
 func GetCloudflowApplicationDescriptorFromDockerImage(dockerRegistryURL string, dockerRepository string, dockerImagePath string) (domain.CloudflowApplicationSpec, docker.PulledImage) {
 
-	apiversion, apierr := exec.Command("docker", "version", "--format", "'{{.Server.APIVersion}}'").Output()
-	if apierr != nil {
-		util.LogAndExit("Could not get docker API version, is the docker daemon running? API error: %s", apierr.Error())
-	}
-
-	trimmedapiversion := strings.Trim(string(apiversion), "\t \n\r'")
-	client, error := docker.GetClient(trimmedapiversion)
-	if error != nil {
-		client, error = docker.GetClient("1.39")
-		if error != nil {
-			fmt.Printf("No compatible version of the Docker server API found, tried version %s and 1.39", trimmedapiversion)
-			panic(error)
-		}
+	client, err := docker.GetVersionedClient()
+	if err != nil {
+		panic(err)
```

Comment:
We should not panic IMHO

Reply:
I think this function should not logAndExit. Only use LogAndExit from the command handling code.

Reply:
done ..